gating/check/pre

```diff
@@ -25,4 +25,25 @@
 # trigger.
 # RE_JOB_REPO_NAME: The project's GitHub repository which gets cloned in the
 # job. For example, for the repository found at
-# https://github.com/rcbops/rpc-gating, the name is rpc-gating.
\ No newline at end of file
+# https://github.com/rcbops/rpc-gating, the name is rpc-gating.
+
+# If the current folder's basename is osp-mnaio then we assume
+# that it is the root of the git clone. If the git clone is not in
+# /opt then we symlink the current folder there so that all the
+# osp-mnaio scripts work as expected.
+if [[ "$(basename ${PWD})" == "osp-mnaio" ]]; then
+  if [[ "${PWD}" != "/opt/osp-mnaio" ]]; then
+    ln -sfn ${PWD} /opt/osp-mnaio
+  fi
+fi
+
+# set osp-mnaio variables
+export OSP_DEPLOY_VERSION=`echo ${RE_JOB_ACTION} | awk -F'_' {'print $2'}`
+export OSP_JOB_ACTION=`echo ${RE_JOB_ACTION} | awk -F'_' {'print $3'}`
+export REDHAT_USERNAME=${RPC_OSP_REDHAT_USERNAME:-''}
+export REDHAT_PASSWORD=${RPC_OSP_REDHAT_PASSWORD:-''}
+export REDHAT_POOL_ID=${RPC_OSP_REDHAT_POOL_ID:-''}
+export REDHAT_ISO_URL=${RPC_OSP_REDHAT_ISO_URL:-''}
+export REDHAT_CONSUMER_NAME="osp-mnaio-PR-test"
+export REDHAT_OSP_VERSION=${OSP_DEPLOY_VERSION:-'13'}
+export REDHAT_OVERCLOUD_REGISTER="false"
```

Comment:
i believe these, if set here in `pre`, are not available in `run` (or `post`) as each script is ran independently within jenkins and therefor best set in a separate file (eg `variables`) that can be sourced by `pre`, `run`, and `post` (if needed).

in this case i believe they are only needed in `run` so they can just be moved there.

Reply:
Yeah, I put them in pre because the comments eluded to that's where configuration options went.  Exporting them should carry them over but yeah it's probably safer just to drop them in run so they've been moved.

Reply:
exporting them tells the shell to carry them over to the environment of any process executed by the current shell, but the current shell only exists for the duration of the script, whether `pre`, `run`, or `post`, and not to the next script which is not executed by the current shell but by groovy.

see https://github.com/rcbops/rpc-gating/blob/master/pipeline_steps/common.groovy#L1462-L1495.